### PR TITLE
VS2017 backend

### PR DESCRIPTION
**Pull request by nioncode**
Adds a backend for VS2017, which partially implements #1390.
I will add auto detection of the VS version, soon.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -20,6 +20,16 @@ environment:
...
       backend: vs2015
 
+    - arch: x86
+      compiler: msvc2017
+      backend: ninja
+      APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+
+    - arch: x86
+      compiler: msvc2017
+      backend: vs2015
```
Comment (msink) [inline .appveyor.yml]: why not vs2017?

File: .appveyor.yml [hunk 2]
```diff
@@ -43,6 +63,7 @@ install:
   - cmd: echo Using Python at %MESON_PYTHON_PATH%
   - cmd: if %compiler%==msvc2010 ( call "C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\vcvarsall.bat" %arch% )
   - cmd: if %compiler%==msvc2015 ( call "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" %arch% )
+  - cmd: if %compiler%==msvc2017 ( call "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat" -arch=%arch% )
```
Comment (msink) [inline .appveyor.yml]: "C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvarsall.bat %arch%"

Comment (nioncode) [inline .appveyor.yml]: Care to elaborate why? vcvarsall just calls VsDevCmd after parsing its arguments, so why don't we just call it directly?

Comment (msink) [inline .appveyor.yml]: Both are okay, really. Matter of taste.

**Comment by msink** [comment]
23 minutes for test on appveyor...

**Comment by nioncode** [comment]
I now added auto detection of the Visual Studio backend.
Closes #1390.

**Comment by jpakkane** [comment]
Is this ready for merging?

**Comment by nioncode** [comment]
All tests pass, so it should be ready 👍

**Pull request merged**